File: Core/YABACore/src/commonMain/kotlin/dev/subfly/yabacore/state/tag/TagCreationUIState.kt
package dev.subfly.yabacore.state.tag

import dev.subfly.yabacore.model.ui.TagUiModel
import dev.subfly.yabacore.model.utils.YabaColor

data class TagCreationUIState(
    val selectedColor: YabaColor = YabaColor.BLUE,
    val selectedIcon: String = "tag-01",
    val label: String = "",
    val editingTag: TagUiModel? = null,
    val isSaving: Boolean = false,
)

File: Darwin/YABACore/State/Creation/Tag/TagCreationUIState.swift
//
//  TagCreationUIState.swift
//  YABACore
//

import Foundation

public struct TagCreationUIState: Sendable {
    public var existingTagId: String?
    public var label: String
    public var icon: String
    public var colorRole: YabaColor

    public init(
        existingTagId: String? = nil,
        label: String = "",
        icon: String = "tag-01",
        colorRole: YabaColor = .none
    ) {
        self.existingTagId = existingTagId
        self.label = label
        self.icon = icon
        self.colorRole = colorRole
    }
}
